routes.test.js

```diff
@@ -0,0 +1,89 @@
+import { Cookie } from 'tough-cookie';
+import { Observable } from 'rxjs';
+import Hapi from 'hapi';
+import getRoutes from './routes';
+import getConfig from './util/config';
+
+const server = new Hapi.Server();
+
+server.connection();
+
+// mock the anonAuthPlugin
+server.decorate(
+	'request',
+	'authorize',
+	request => () => Observable.of(request),
+	{ apply: true }
+);
+
+const MOCK_RESULT = 'asdf';
+const MOCK_RENDER_REQUEST = () => Observable.of({ result: MOCK_RESULT, statusCode: 200 });
+
+const MOCK_renderRequestMap = {
+	'en-US': MOCK_RENDER_REQUEST,
+};
+
+const MOCK_OAUTH_COOKIES = {
+	oauth_token: '1234',
+	refresh_token: 'asdf',
+	anonymous: true
+};
+
+const MOCK_API_RESULT = {
+	queries: [],
+	responses: [],
+};
+
+const serverReady = getConfig()
+	.then(config => getRoutes(MOCK_renderRequestMap, config, () => () => Observable.of(MOCK_API_RESULT)))
+	.then(server.route.bind(server));
+
+describe('routes', () => {
+	it('serves the homepage route', () =>
+		serverReady
+			.then(() => server.inject({ url: '/' }))
+			.then(response => expect(response.payload).toEqual(MOCK_RESULT))
```

Comment:
Yeah the tools for mocking Hapi requests are a little opaque, unfortunately, but basically `server.inject` on the previous line creates a fake [Hapi request object](http://hapijs.com/api#request-object), which allows you to inspect the response of your routes for the expected features.

I'll add a bunch of code comments here before I pull off the `Prelim` label